staging/src/k8s.io/apiserver/pkg/registry/rest/meta.go

```diff
@@ -28,7 +28,6 @@ import (
 func FillObjectMetaSystemFields(meta metav1.Object) {
 	meta.SetCreationTimestamp(metav1.Now())
 	meta.SetUID(uuid.NewUUID())
-	meta.SetSelfLink("")
```

Comment:
should we not continue clearing any user-provided selfLink value here?

Reply:
Hmm - TBH I don't understand what's the point of this row. "" is a default value for a string (as opposed to the other two above for timestamp and uid), so I don't understand what it is supposed to give us.

Reply:
This ensures any user-specified value in self link is cleared. We should avoid any visible behavior change to any actor as a result of this PR. I think dropping this here would let a user-specified self-link be exposed to an admission webhook (I think the PrepareObjectForStorage call happens after webhooks are called)

Reply:
Fair - reverted.